Gems/Compression/Code/Source/Clients/DecompressionRegistrarImpl.h

```diff
@@ -0,0 +1,85 @@
+        AZ::Outcome<void, AZStd::unique_ptr<IDecompressionInterface>> RegisterDecompressionInterface(
+            CompressionAlgorithmId algorithmId,
+            AZStd::unique_ptr<IDecompressionInterface> decompressionInterface) override;
+
+        //! Registers a decompression interface with a null deleter
+        bool RegisterDecompressionInterface(CompressionAlgorithmId algorithmId, IDecompressionInterface& decompressionInterface) override;
+
+        bool UnregisterDecompressionInterface(CompressionAlgorithmId algorithmId) override;
+
+        [[nodiscard]] IDecompressionInterface* FindDecompressionInterface(CompressionAlgorithmId algorithmId) const override;
+
+        [[nodiscard]] bool IsRegistered(CompressionAlgorithmId algorithmId) const override;
+
+        struct DecompressionInterfaceDeleter
+        {
+            DecompressionInterfaceDeleter();
+            DecompressionInterfaceDeleter(bool shouldDelete);
+            void operator()(IDecompressionInterface* ptr) const;
+            bool m_delete{ true };
+        };
+    private:
+        using DecompressionInterfacePtr = AZStd::unique_ptr<IDecompressionInterface, DecompressionInterfaceDeleter>;
+
+        //! Helper function that is used to register a decompression interface into the decompression interface array
+        //! while taking into account whether the decompression interface should be owned by this registrar
+        //! @param compressionAlgorithmId Unique Id of the decompression interface to register with this registrar
+        //! @param decompressionInterface unique_ptr to decompression interface to register
+        //! @return outcome which indicates whether the decompression interface was registered with the decompression interface array
+        //! On success an empty Success outcome is returned
+        //! On failure, the supplied decompression interface parameter is returned back to the caller
+        [[nodiscard]] AZ::Outcome<void, DecompressionInterfacePtr> RegisterDecompressionInterfaceImpl(
+            CompressionAlgorithmId algorithmId, DecompressionInterfacePtr decompressionInterface);
+
+        struct DecompressionIdIndexEntry
+        {
+            CompressionAlgorithmId m_id;
+            DecompressionInterfacePtr m_decompressionInterface;
+        };
+
+        using IdToDecompressionInterfaceMap = AZStd::vector<DecompressionIdIndexEntry>;
+
+        //! Searches within the decompression interface array for the decompression interface registered with the specified id
+        //! @param algorithmId Unique Id of decompression interface to locate
+        //! @return iterator pointing the decompression interface registered with the specified CompressionAlgorithmId
+        //! NOTE: It is responsibility of the caller to lock the decompression interface mutex to protect the search
+        typename IdToDecompressionInterfaceMap::const_iterator FindDecompressionInterfaceImpl(CompressionAlgorithmId) const;
```

Comment:
Should match doxygen entry for ```algorithmId```
```suggestion
        typename IdToDecompressionInterfaceMap::const_iterator FindDecompressionInterfaceImpl(CompressionAlgorithmId algorithmId) const;
```